cms/static/js/utils/date_utils.js

```diff
@@ -22,19 +30,31 @@ function($, date, TriggerChangeEventOnEnter) {
         } else if (selectedDate) {
             return new Date(Date.UTC(
                 selectedDate.getFullYear(), selectedDate.getMonth(), selectedDate.getDate()));
+        } else if (selectedRelative) {
+            return parseFloat(selectedRelative);
         } else {
             return null;
         }
     }
 
-    function setDate(datepickerInput, timepickerInput, datetime) {
+    function setDate(datepickerInput, timepickerInput, relativeTimeInput, datetime) {
+        var parsedRelativeDate, parsedDatetime;
+
         // given a pair of inputs (datepicker and timepicker) and the date as an
-        // ISO-formatted date string.
-        var parsedDatetime = Date.parse(datetime);
-        if (parsedDatetime) {
-            $(datepickerInput).datepicker('setDate', parsedDatetime);
-            if (timepickerInput.length > 0) {
-                $(timepickerInput).timepicker('setTime', parsedDatetime);
+        // ISO-formatted date string or a relative date formatted as +FLOAT or -FLOAT
+        // where INT is a number of seconds
+        if (datetime && typeof datetime === 'number') {
```

Comment:
seems odd to have a number type stored in a var named `datetime`.  maybe change that function arg to something like `dateValue`?